osf/models/registrations.py

```diff
@@ -469,7 +469,10 @@ def is_pending_review(self):
     def is_approved(self):
         if self.requires_approval:
             if not self.approval:
-                return False
+                if self.registered_node is None:
```

Comment:
`get_approval_status` already has a check for `if draft.registered_node is not None`, but it occurs after `if draft.is_approved` which means it will never hit the 'approved but canceled', right?

If you can just rework the get_approval_status without changing how the model works, I think that would be desirable.

Reply:
Um...I don't understand. If `draft.is_approved==True`, then it could mean either `draft.registered_node is not None and draft.approval is None` or `draft.approval.is_approved==True`. If it is the first situation, then it means it would go on and check whether `draft.registered_node.isdeleted==True`. If yes then `approved but canceled` would be hit.
Realistically, I could modify the `get_approval_status` method only (by not using the `is_approved` property at all) without addressing the inherent problem that the `is_approved` method reflects the wrong admin approval status of a draft registration. In that case, there could be weird things happening like `get_approval_status` returns 'Approved but canceled' while `is_approved` returns `False`......

Reply:
Discussed briefly in person, I *think* that this change won't get hit. Either way, tests will clarify the state 👍